pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -502,11 +530,28 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 			changes.CreateSandbox = false
 			return changes
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.SidecarLifecycle) {
+			if pod.Spec.RestartPolicy != v1.RestartPolicyAlways && onlySidecars(pod, podStatus) && len(podStatus.ContainerStatuses) != 0 {
+				changes.CreateSandbox = false
+				return changes
+			}
+		}
 		if len(pod.Spec.InitContainers) != 0 {
 			// Pod has init containers, return the first one.
 			changes.NextInitContainerToStart = &pod.Spec.InitContainers[0]
 			return changes
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.SidecarLifecycle) {
+			// If pod has sidecars, start them before non sidecars
+			sidecarContainerIDs := getSidecarIDs(pod)
```

Comment:
99% sure this is the case but just double checking... we do not need to be worried about a race condition here, because idx is a stable identifier, given that pod.Spec.Containers can't be updated?

Reply:
yeah their position is immutable, if that changed it would be a new pod.